src/convertToObject.js

```diff
@@ -6,7 +6,17 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const res = sourceString.split(';').reduce((acc, str) => {
+    if (str.trim()) {
+      const [key, value] = str.split(':');
```

Comment:
Consider adding a check to ensure that the `str` contains a colon before attempting to split it into `key` and `value`. This will prevent potential errors if the input string does not follow the expected format.